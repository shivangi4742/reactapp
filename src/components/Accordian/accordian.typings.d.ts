import { ReactNode } from "react";
import { string } from "prop-types";


export type AccordianheaderTypes = {
    header?: string
    children: ReactNode,
    type: string
}
