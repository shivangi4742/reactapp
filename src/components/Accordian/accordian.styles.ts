import styled, { css } from 'styled-components';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import DS from '../../utils/design-system';

export const ExpansionPanelheaderstyle = styled(ExpansionPanelSummary)`

    ${props => props.type == 'primary' && css`
        background: ${DS.PALETTE.BLACK}!important;
        border: 1px solid transparent;
        color: ${DS.PALETTE.WHITE}!important;
   `}
`;

export const Left = styled.span`
  float : left;
`;
export const Right = styled.span`
  float : right
`;
