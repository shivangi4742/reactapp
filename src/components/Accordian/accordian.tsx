
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import React, { ReactNode, Children } from 'react';
import { ExpansionPanelheaderstyle } from './accordian.styles';
import ExpansionPanelActions from '@material-ui/core/ExpansionPanelActions';
import Divider from '@material-ui/core/Divider';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';

type AccordianType = {
    header?: ReactNode,
    footer?: ReactNode,
    children?: ReactNode,
    disabled?: boolean,
    type?: string
}
function Accordian({ header, footer, children, ...props }: AccordianType) {
    console.log(props);
    return (
        <ExpansionPanel style={{ marginBottom: '12px' }} {...props} >
            <ExpansionPanelheaderstyle   {...props}>
                {header}
            </ExpansionPanelheaderstyle>
            <ExpansionPanelDetails>
                {children}
            </ExpansionPanelDetails>
            {footer ? (<div> <Divider />
                <ExpansionPanelActions>
                    {footer}
                </ExpansionPanelActions></div>) : null}
        </ExpansionPanel >

    );
}


export default Accordian;
