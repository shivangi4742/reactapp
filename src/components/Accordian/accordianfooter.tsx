import React from 'react';

const AccordianFooter = (props) => {
    return (
        <div>
            {props.children}
        </div>
    )
}

export default AccordianFooter;