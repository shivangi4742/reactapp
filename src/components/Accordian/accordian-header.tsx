import React from 'react';
import { Right, Left } from './accordian.styles';
import Grid from '@material-ui/core/Grid';

const Accordianheader = (props) => {
    return (
        <Grid container >
            <Grid item sm={props.right ? (6) : (12)} xs={6}>
                <Left>{props.left} {' '}{props.title}</Left>
            </Grid>
            <Grid item sm={6} xs={6}>
                <Right>{props.right}</Right>
            </Grid>
        </Grid>
    )
}

export default Accordianheader;