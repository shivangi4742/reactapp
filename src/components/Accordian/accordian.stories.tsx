import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Accordian from './accordian';
import Accordianheader from './accordian-header';
import { Grid } from '@material-ui/core';
import ComponentAPI from '../component-api';
import Button from '../Button/button';
import ComponentStories from '../component-stories';
import Storiesheader from '../stories-header';
const stories = storiesOf('Components', module);

const codeString = `
 <Accordian header={<Accordianheader title="hello" type="primary" />}>
          add child components.
</Accordian>
`;
const codeString1 = `
 <Accordian header={<Accordianheader title="hello" />} >
          add child components.
 </Accordian>
`;
const codeString2 = `
<Accordian header={<Accordianheader title="hello" />} footer={<Button>save</Button>} >
        add child components.
</Accordian>
`;
const codeString3 = `
<Accordian header={<Accordianheader title="hello" />} disabled>
        add child components.
</Accordian>
`;

const rows = [
    {
        id: 1,
        property: 'type',
        description: 'can be set to primary for black header',
        type: 'string',
        default: "-"
    },
    {
        id: 2,
        property: 'disabled',
        description: 'can be set to disabled',
        type: 'boolean',
        default: "-"
    },
    {
        id: 2,
        property: 'header, footer',
        description: 'can be pass header, footer & childcomponents',
        type: 'components',
        default: "-"
    }
]
stories.add(
    'Accordian',
    (() => <div>
        <Storiesheader desc="Accordion is a special kind of Collapse, which allows only one panel to be expanded at a time." shortdesc="A content area which can be collapsed and expanded." heading="Accordian" />
        <h3>Examples <a href="#examples">#</a></h3>

        <Grid container spacing={40}>
            <ComponentStories syntax={codeString} >
                <Accordian header={<Accordianheader title="hello" />} type="primary" >
                    add child components.
                </Accordian>
            </ComponentStories>
            <ComponentStories syntax={codeString1} >
                <Accordian header={<Accordianheader title="hello" />} >
                    add child components.
                </Accordian>
            </ComponentStories>
            <ComponentStories syntax={codeString2} >
                <Accordian header={<Accordianheader title="hello" />} footer={<Button>save</Button>} >
                    add child components.
                 </Accordian>
            </ComponentStories>
            <ComponentStories syntax={codeString3} >
                <Accordian header={<Accordianheader title="hello" />} disabled >
                    add child components.
                 </Accordian>
            </ComponentStories>
        </Grid>
        <ComponentAPI rows={rows} />
    </div >
    )
);

