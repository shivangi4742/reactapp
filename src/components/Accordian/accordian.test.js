import React from 'react';
import { shallow } from 'enzyme';
import Accordian from './index';
import toJson from 'enzyme-to-json';

describe('Accordian', () => {
   
    it('is defined', () => {
        const accordian = shallow(<Accordian />);
        expect(accordian).toBeDefined();
    });
    it('matches the snapshots of Accordian component', () => {
        const tree = shallow(<Accordian />);
        expect(toJson(tree)).toMatchSnapshot();
    });
})