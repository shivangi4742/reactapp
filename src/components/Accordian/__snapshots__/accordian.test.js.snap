// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian matches the snapshots of Accordian component 1`] = `
<Styled(WithStyles(ExpansionPanel))>
  <Styled(WithStyles(ExpansionPanelSummary)) />
  <WithStyles(ExpansionPanelDetails) />
</Styled(WithStyles(ExpansionPanel))>
`;
